components/feature/pwa/src/main/java/mozilla/components/feature/pwa/intent/WebAppIntentProcessor.kt

```diff
@@ -61,6 +65,34 @@ class WebAppIntentProcessor(
         }
     }
 
+    /**
+     * Returns an existing web app session that matches the manifest.
+     */
+    private fun findExistingSession(webAppManifest: WebAppManifest): Session? {
+       return sessionManager.all.find {
+           it.customTabConfig?.externalAppType == ExternalAppType.PROGRESSIVE_WEB_APP &&
+                   it.webAppManifest?.startUrl == webAppManifest.startUrl
+       }
+    }
+
+    /**
+     * Returns a new web app session.
+     */
+    private fun createSession(webAppManifest: WebAppManifest, url: String): Session {
+        return Session(url, private = false, source = Source.HOME_SCREEN)
+                .apply {
+                    this.webAppManifest = webAppManifest
+                    this.customTabConfig = webAppManifest.toCustomTabConfig()
+                }.also {
+                    sessionManager.add(it)
+                    loadUrl(url, it)
+                }
+    }
+
+    private fun loadUrl(url: String, session: Session) {
```

Comment:
I don't feel like `loadUrl` is needed, we can just keep `loadUrlUseCase` inlined in `createSession`